MSAL/src/instance/MSALAccountsProvider.m

```diff
@@ -116,8 +122,17 @@ - (MSALAccount *)accountForHomeAccountId:(NSString *)homeAccountId
 
     if ([msidAccounts count])
     {
-        MSALAccount *msalAccount = [[MSALAccount alloc] initWithMSIDAccount:msidAccounts[0]];
-        return msalAccount;
+        NSArray<MSALAccount *> *msalAccounts = [self msalAccountsFromMSIDAccounts:msidAccounts];
+        if (msalAccounts.count == 1)
+        {
+            return msalAccounts[0];
+        }
+        else if (msalAccounts.count > 1)
+        {
+            MSID_LOG_WARN(nil, @"Retrieved more than 1 msal accounts for the same home accout id! (More info: environments are equal for first 2 accounts: %@, usernames are equal for first 2 accounts: %@)", msalAccounts[0].environment == msalAccounts[1].environment ? @"YES" : @"NO", msalAccounts[0].username == msalAccounts[1].username? @"YES" : @"NO");
```

Comment:
Isn't this one expected?

Reply:
Not really, when home account id is provided, we expect one MSAL account but with multiple tenant profiles :)